src/test/compile-fail/lint-type-overflow2.rs

```diff
@@ -9,16 +9,18 @@
 // except according to those terms.
 //
 
-#![deny(overflowing_literals)]
-#![deny(const_err)]
+#![warn(overflowing_literals)]
```

Comment:
I think what happened is that errors were emitted earlier in compilation, and then presumably something had `abort_if_errors` which aborted compilation before the next lint location was reached. Before this PR all those errors were buffered until the end, but now that they're emitted on-demand they presumably cause compilation to abort halfway through (or somethign like that)